src/tests/test_datastructures.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (C) 2021 NV Access Limited
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+from src.transform.datastructures import NVDAVersion
+import unittest
+
+
+class TestNVDAVersion(unittest.TestCase):
+	"""
+	Ensure that if a patch number is specified, it is preserved in strings, otherwised ignored.
+	For example NVDAVersion(2019, 1) should not end up creating paths like /2019.1.0/.
+	"""
+	def test_compare(self):
+		"""Test comparing versions"""
+		self.assertEqual(NVDAVersion(13, 2, 0), NVDAVersion(13, 2))
+		self.assertLess(NVDAVersion(13, 2), NVDAVersion(13, 2, 1))
+		self.assertGreater(NVDAVersion(3, 2, 1), NVDAVersion(1, 2, 3))
+
+	def test_toStr(self):
+		"""Test converting versions to string"""
+		self.assertEqual(NVDAVersion(1, 3, 4).toStr(), "1.3.4")
+		self.assertEqual(NVDAVersion(13, 2).toStr(), "13.2")
+		self.assertEqual(NVDAVersion(13, 2, 0).toStr(), "13.2.0")
+
+	def test_fromStr(self):
```

Comment:
Using three integers for versions in the `nvdaVersion.json` file would allow deleting all the "fromStr" code and tests.

Reply:
I've done this